src/reformat.c

```diff
@@ -738,48 +815,58 @@ static avifResult avifImageYUVAnyToRGBAnySlow(const avifImage * image,
-                    const int Co = (int)avifRoundf(Cr * yuvMaxChannel);
-                    const int t = YY - (Cg >> 1);
-                    G = (float)AVIF_CLAMP(t + Cg, 0, state->rgb.maxChannel);
-                    B = (float)AVIF_CLAMP(t - (Co >> 1), 0, state->rgb.maxChannel);
-                    R = (float)AVIF_CLAMP(B + Co, 0, state->rgb.maxChannel);
-                    G /= rgbMaxChannelF;
-                    B /= rgbMaxChannelF;
-                    R /= rgbMaxChannelF;
+            float Rc = 0.0f, Gc = 0.0f, Bc = 0.0f, grayc = 0.0f;
+            if (rgbHasColor) {
+                float R, G, B;
+                if (yuvHasColor) {
+                    if (state->yuv.mode == AVIF_REFORMAT_MODE_IDENTITY) {
+                        // Identity (GBR): Formulas 41,42,43 from
+                        // https://www.itu.int/rec/T-REC-H.273-201612-S
+                        G = Y;
+                        B = Cb;
+                        R = Cr;
+                    } else if (state->yuv.mode == AVIF_REFORMAT_MODE_YCGCO) {
+                        // YCgCo: Formulas 47,48,49,50 from
+                        // https://www.itu.int/rec/T-REC-H.273-201612-S
+                        const float t = Y - Cb;
+                        G = Y + Cb;
+                        B = t - Cr;
+                        R = t + Cr;
+                    } else if ((state->yuv.mode == AVIF_REFORMAT_MODE_YCGCO_RE) || (state->yuv.mode == AVIF_REFORMAT_MODE_YCGCO_RO)) {
+                        // YCgCoRe/YCgCoRo: Formulas 62,63,64,65 from
+                        // https://www.itu.int/rec/T-REC-H.273-202407-P
+                        const int YY = unormY;
+                        const int Cg = (int)avifRoundf(Cb * yuvMaxChannel);
+                        const int Co = (int)avifRoundf(Cr * yuvMaxChannel);
+                        const int t = YY - (Cg >> 1);
+                        G = (float)AVIF_CLAMP(t + Cg, 0, state->rgb.maxChannel);
+                        B = (float)AVIF_CLAMP(t - (Co >> 1), 0, state->rgb.maxChannel);
+                        R = (float)AVIF_CLAMP(B + Co, 0, state->rgb.maxChannel);
+                        G /= rgbMaxChannelF;
+                        B /= rgbMaxChannelF;
+                        R /= rgbMaxChannelF;
+                    } else {
+                        // Normal YUV
+                        R = Y + (2 * (1 - kr)) * Cr;
+                        B = Y + (2 * (1 - kb)) * Cb;
+                        G = Y - ((2 * ((kr * (1 - kr) * Cr) + (kb * (1 - kb) * Cb))) / kg);
+                    }
                 } else {
-                    // Normal YUV
-                    R = Y + (2 * (1 - kr)) * Cr;
-                    B = Y + (2 * (1 - kb)) * Cb;
-                    G = Y - ((2 * ((kr * (1 - kr) * Cr) + (kb * (1 - kb) * Cb))) / kg);
+                    // Monochrome: just populate all channels with luma (state->yuv.mode
+                    // is irrelevant)
```

Comment:
Nit: If the matrix coefficients is 0 (identity matrix), we should perform RGB-to-YUV conversion and then take the luma output. I don't know what matrix coefficients we should use in this RGB-to-YUV conversion...